src/utils/DotnetDevCertsHttps.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { join } from "path";
+import { execChildProcess } from "../common";
+import { CoreClrDebugUtil } from "../coreclr-debug/util";
+
+// Will return true if `dotnet dev-certs https --check` succesfully finds a development certificate. 
+export async function hasDotnetDevCertsHttps(dotNetCliPaths: string[]): Promise<Boolean> {
+
+    let dotnetExecutablePath = getDotNetExecutablePath(dotNetCliPaths);
+
+    try {
+        const data = await execChildProcess(`${dotnetExecutablePath ?? 'dotnet'} dev-certs https --check`, process.cwd(), process.env);
+        return (/A valid certificate was found/.test(data));
+    }
+    catch (err) {
+        return false;
+    }
+}
+
+// Will run `dotnet dev-certs https --trust` to prompt the user to create self signed certificates. Retruns true if sucessfull.
+export async function createSelfSignedCert(dotNetCliPaths: string[]): Promise<Boolean> {
```

Comment:
>Boolean [](http://example.com/codeflow?start=78&length=7)

I think I would instead suggest returning an error string if this fails rather than just a `Boolean`. I would suggest including the stderr and exit code.

Reply:
Should I show the error string in the warning message when it fails?